wa-integrations/teamsbot/routes/bot.js

```diff
@@ -0,0 +1,144 @@
+  // Send a trace activity, which will be displayed in Bot Framework Emulator
+  await turnContext.sendTraceActivity('OnTurnError Trace', error, 'https://www.botframework.com/schemas/error', 'TurnError');
+
+  // Send a message to the user
+  await turnContext.sendActivity('The bot encountered an error or bug.');
+  await turnContext.sendActivity('To continue to run this bot, please fix the bot source code.');
+};
+
+var teamsUsers = new hashMap(); // List of all registered teams users
+
+// Common function for sending a message to WhatsApp
+async function sendWAMessage(strWhatsAppNumber, strText, res) {
+  const tyntecHeaders = {
+    'Content-Type': 'application/json',
+    apikey: appConfig.TYNTEC_API_KEY,
+  };
+
+  const strURL = 'https://api.tyntec.com/chat-api/v2/messages';
+  const inputBody = {
+    to: strWhatsAppNumber,
+    channels: ['whatsapp'],
+    whatsapp: {
+      from: appConfig.TYNTEC_WABA_NUMBER,
+      contentType: 'text',
+      text: strText,
+    },
+  };
+
+  try {
+    await axios.post(strURL, inputBody, { headers: tyntecHeaders });
+    res.sendStatus(204);
+  } catch (e) {
+    handleError(e);
+    res.sendStatus(500);
+  }
+}
+
+// Teams -> WhatsApp
+router.post('/forwardwhatsapp', function (req, res, next) {
+  botAdapter.processActivity(req, res, async (turnContext) => {
+    if (turnContext.activity.type === 'message') {
+      // Check if user is already known
+      if (teamsUsers.has(turnContext.activity.from.id) == false) {
+        // If not, Add the teams user to list of all users and set availability to true
+        console.log('First time this teams user is recognized.');
+
+        var userDetail = {};
+        userDetail.available = true;
+        userDetail.conversationReference = TurnContext.getConversationReference(turnContext.activity);
+        userDetail.assignedWhatsAppNumber = 0;
+        teamsUsers.set(turnContext.activity.from.id, userDetail);
+
+        var strMessage =
+          'Hey, welcome to the bot. Your teams user ID: **' +
+          turnContext.activity.from.id +
+          '**. You will now receive messages from WhatsApp.';
+        await turnContext.sendActivity(strMessage);
+
+        console.log('All teams users:\n');
+        console.log(teamsUsers);
+      } else {
+        console.log('Send input from teams user to WhatsApp.');
+        console.log('Received teams message: ', turnContext.activity.text);
+        console.log('From user: ', turnContext.activity.from.id);
+
+        await sendWAMessage(teamsUsers.get(turnContext.activity.from.id).assignedWhatsAppNumber, turnContext.activity.text, res);
+      }
+
+      next();
+    }
+  });
+});
+
+// WhatsApp -> Teams
+router.post('/forwardteams', async (req, res) => {
+  console.log('Send input from WhatsApp to teams user.');
+  console.log('Received WA message: ', req.body.content.text);
+  console.log('From number: ', req.body.from);
+  var strWhatsAppNumber = req.body.from;
```

Comment:
First time I see the the type on the variable present.

Couldn't this be simply `whatsAppNumber`?